light-client/src/types.rs

```diff
@@ -117,6 +139,27 @@ impl LightBlock {
     pub fn height(&self) -> Height {
         self.signed_header.header.height
     }
+
+    /// Obtain the verification parameters for the light block when using it as
+    /// trusted state.
+    pub fn as_trusted_state(&self) -> TrustedBlockState<'_> {
```

Comment:
Not with the way the `TrustedBlockState` and `UntrustedBlockState` structs are defined, since they take references to (potentially large) data structures specifically to avoid having to clone those data structures.